src/chrome/inject.js

```diff
@@ -1,46 +1,113 @@
+    window.addEventListener('scroll', handleScroll)
+    handleScroll()
+})()
+
+
+
+function dumpAvgConfidence(product){
+    const confidenceDOM = document.createElement("div");
+    confidenceDOM.style.marginTop = '.5em'
+    const pro = calculations.getProportionPositiveFromAvg(product.rating)
+    const ci = calculations.evaluateRatings(pro, product.reviewCount)
+    confidenceDOM.innerHTML = `AVG CI for ${product.rating}, n=${product.reviewCount} is <br> proportion: ${ci.proportion} <br> lower: ${ci.lower} <br> upper: ${ci.upper}`;
+    product.dom.insertAdjacentElement("beforeend", confidenceDOM);
+}
+
+
+/**
+ * For each product, create a beta distribution visualization based on average rating
+ * and insert it into the DOM
+ * @param {Product} product
+ */
+function insertAvgBetaViz(product){
+    // Create color scale
+    const cmap = "interpolateSpectral";
+    const confidenceDOM = document.createElement("div");
+    const proportion = calculations.getProportionPositiveFromAvg(product.rating);
+    const betaParams = calculations.getBetaParams(proportion, product.reviewCount);
+    betaviz.addViz(confidenceDOM, betaParams, cmap);
+    product.dom.insertAdjacentElement("beforeend", confidenceDOM);
+}
+
+
 /**
- * This script may run more than one time within the same context (e.g, if use goes to another results page)
- * To avoid namespace issues, we check if window.injectConfidenceInterval already exists (meaning the
- * script has run before and doesn't need to be re-declared)
+ * For each product, create a beta distribution visualization based on distribution
+ * and insert it into the DOM
+ * @param {Product} product
  */
-if (!window.injectConfidenceInterval) {
-    window.injectConfidenceInterval = async () => {
-        const productsSelector =
-            "div.a-section.a-spacing-none.a-spacing-top-micro>.a-row.a-size-small";
+function insertDistBetaViz(product){
+    // Create color scale
+    const cmap = "interpolateSpectral";
+    const confidenceDOM = document.createElement("div");
+    const proportion = calculations.getProportionPositiveFromDist(product.distributions);
+    const betaParams = calculations.getBetaParams(proportion, product.reviewCount);
+    betaviz.addViz(confidenceDOM, betaParams, cmap);
+    product.dom.insertAdjacentElement("beforeend", confidenceDOM);
+}
 
-        let productsDOM = document.querySelectorAll(productsSelector);
 
-        // Wait until product ratings have been loaded into the DOM
-        while (!productsDOM.length) {
-            await new Promise(r => setTimeout(r, 500));
-            productsDOM = document.querySelectorAll(productsSelector);
+function handleScroll(evt){
+    if(scrollTicking)  return
+    scrollTicking = true
+    window.requestAnimationFrame(()=>{
+        const leftover = []
+        for(let product of noDistProducts) {
+            if (productInView(product)){
+                // noinspection JSIgnoredPromiseFromCall
```

Comment:
shouldn't you also check for `product.reviews`? I noticed that in products without reviews, the avg viz doesn't get added, but a blank dist viz does

Reply:
Good find! I've pushed an update that filters out those products before setting `noDistProducts`